docs/guides/tfhers.md

```diff
@@ -0,0 +1,242 @@
+# TFHE-rs Compatibility
+
+This document explains how to combine Concrete and [TFHE-rs](https://github.com/zama-ai/tfhe-rs) computations together. This feature targets two main use cases.
+
+## Table of Content
+
+- [The Basics](#the-basics)
+- [First Use Case](#first-use-case-shared-secret-key)
+- [Second Use Case](#second-use-case-pregenerated-tfhe-rs-keys)
+- [Serialization of Ciphertexts and Keys](#serialization-of-ciphertexts-and-keys)
+
+## The Basics
+
+There are differences between Concrete and TFHE-rs, so ensuring compatibility between them involves more than just data serialization. To achieve compatibility, we need to consider two main aspects.
+
+1. In TFHE-rs and Concrete, integers are encoded differently. Both use LWE ciphertexts, but they organize them differently to represent integers. In Concrete, TFHE-rs big integers are represented as a vector of `n` Concrete integers. `n` is determined by dividing the TFHE-rs integer bit-width by the TFHE-rs message bit-width. These bit-width values, along with the carry bit-width, are the encoding parameters used in Concrete to represent TFHE-rs big integers. When working with a TFHE-rs integer type in Concrete, you can use the `.encode(...)` and `.decode(...)` functions to see this in practice.
+
+2. The Concrete Optimizer doesn't limit its search space to available TFHE-rs parameters. Therefore, we need to either fix or limit the search space in areas of the circuit where compatibility is required. This will ensure that the same parameters are used where needed.
```

Comment:
Proposal: The Concrete Optimizer doesn't limit its search space to available TFHE-rs parameters.  --> The Concrete Optimizer can find parameters which are not in TFHE-rs search space.  (or, which are not in TFHE-rs pre-computed list of parameters. )